inference-engine/tests/functional/plugin/shared/src/single_layer_tests/tensor_iterator.cpp

```diff
@@ -0,0 +1,230 @@
+                 targetDevice) = obj.param;
+        std::vector<std::vector<size_t>> inputShapes = {};
+
+        std::string type;
+        switch (ti_body) {
+            case TensorIteratorBody::LSTM:
+                type = "LSTM";
+                inputShapes = {
+                        {{batch, input_size}, {batch, hidden_size}, {batch, hidden_size}, {4 * hidden_size, input_size},
+                                {4 * hidden_size, hidden_size}, {4 * hidden_size}},
+                };
+                break;
+            case TensorIteratorBody::GRU:
+                type = "GRU";
+                inputShapes = {
+                        {{batch, input_size}, {batch, hidden_size}, {3 * hidden_size, input_size},
+                                {3 * hidden_size, hidden_size}, {3 * hidden_size}},
+                };
+                break;
+            case TensorIteratorBody::RNN:
+                type = "RNN";
+                inputShapes = {{batch, input_size}, {batch, hidden_size},
+                               {hidden_size, input_size}, {hidden_size, hidden_size}, {hidden_size}};
+                break;
+        }
+
+        std::ostringstream result;
+        result << "unrolling=" << should_decompose << "_";
+        result << "seq_lenghts=" << seq_lenghts << "_";
+        result << "batch=" << batch << "_";
+        result << "hidden_size=" << hidden_size << "_";
+        result << "input_size=" << input_size << "_";
+        result << "IS=" << CommonTestUtils::vec2str(inputShapes) << "_";
+        result << "TensorIteratorBody=" << type << "_";
+        result << "direction=" << direction << "_";
+        result << "clip=" << clip << "_";
+        result << "netPRC=" << netPrecision.name() << "_";
+        result << "targetDevice=" << targetDevice << "_";
+        return result.str();
+    }
+
+    void TensorIteratorTest::SetUp() {
+        size_t seq_lenghts;
+        bool should_decompose;
+        size_t batch;
+        size_t hidden_size;
+        size_t input_size;
+        TensorIteratorBody ti_body;
+        float clip;
+        ngraph::op::RecurrentSequenceDirection direction;
+        InferenceEngine::Precision netPrecision;
+        std::tie(should_decompose, seq_lenghts, batch, hidden_size, input_size, clip, ti_body, direction, netPrecision,
+                 targetDevice) = this->GetParam();
+        std::vector<std::vector<size_t>> inputShapes;
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+        auto tensor_iterator = std::make_shared<ngraph::opset5::TensorIterator>();
+
+        // Each case consist of 3 steps:
+        // 1. Create TensorIterator body.
+        // 2. Set PortMap
+        // 3. Create outer function
+        auto axis = std::make_shared<ngraph::opset5::Constant>(ngraph::element::i64, ngraph::Shape{1}, std::vector<int64_t>{1});
+        switch (ti_body) {
+            case TensorIteratorBody::LSTM: {
+                inputShapes = {
```

Comment:
Please, move ti creation logic to the [`builders`](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/ngraph_functions/include/ngraph_functions/builders.hpp)

Reply:
Not in this PR, have no time for it, not a blocker. Please create a separate ticket, if necessary